src/targets/gpu/kernels/include/migraphx/kernels/index.hpp

```diff
@@ -36,44 +36,33 @@ namespace migraphx {
 #if defined(MIGRAPHX_NGLOBAL) && defined(MIGRAPHX_NLOCAL)
 #define MIGRAPHX_NGROUP ((MIGRAPHX_NGLOBAL + MIGRAPHX_NLOCAL - 1) / MIGRAPHX_NLOCAL)
 #endif
+inline __device__ __attribute__((const)) index_int compute_max_local_size()
+{
+#ifdef MIGRAPHX_NLOCAL
+    return MIGRAPHX_NLOCAL;
+#else
+    MIGRAPHX_ASSERT(compute_global_size() % blockDim.x == 0);
+    return blockDim.x;
+#endif
+}
 
 inline __device__ __attribute__((const)) index_int compute_global_size()
 {
 #ifdef MIGRAPHX_NGLOBAL
     return MIGRAPHX_NGLOBAL;
 #else
-    // This actualy works even when global is not divisible by local size.
-    // This doesnt actually do a multiplicatiosn. Instead it calls a device
-    // function to get the global size, which is why it works.
-    return blockDim.x * gridDim.x;  // NOLINT
+    return compute_max_local_size() * gridDim.x; // NOLINT
```

Comment:
This is wrong, this will compute the wrong global size when using non-uniform tiles. It should be `blockDim.x * gridDim.x`(which calls `__ockl_get_global_size` see [here](https://github.com/ROCm-Developer-Tools/clr/blob/5914ac3c6e9b3848023a7fa25e19e560b1c38541/hipamd/include/hip/amd_detail/amd_hip_runtime.h#L324C10-L324C35)) or we could call `__ockl_get_global_size(0)` directly.

Also, this creates an infinite recursion because `compute_max_local_size` will call `compute_global_size` when assertions are enabled.

Reply:
> This is wrong, this will compute the wrong global size when using non-uniform tiles. 

For the non-uniform block `blockDim.x` would return dimension of non-uniform block and it wouldn't calculate global size correctly.